openedx/core/djangoapps/content/block_structure/tests/test_signals.py

```diff
@@ -69,3 +71,14 @@ def test_course_delete(self):
             bs_manager.get_collected()
 
         self.assertFalse(is_course_in_block_structure_cache(self.course.id, self.store))
+
+
+class CourseBlocksSignalHandlerTest(ModuleStoreTestCase):
+    """
+    Tests specific to the course blocks signal handlers.
+    """
+    @patch('openedx.core.djangoapps.content.block_structure.tasks.update_course_in_cache.apply_async')
+    def test_library_update(self, mock_update):
+        library_key = LibraryLocator(org='org', course='course')
+        _listen_for_course_publish(sender=None, course_key=library_key)
+        self.assertFalse(mock_update.called)
```

Comment:
Nit: To make sure that your test works, consider DDTing the `course_key` and then make sure the `mock` gets called for `CourseLocator` and not for `LibraryLocator`.